### PR TITLE
Deploy to staging on AWS

### LINKED ISSUE #8: Set up deployment on staging server(s)
## Goal

Full stack deployment of LANE on Heroku or other services

## Acceptance criteria
- [x] Server deployed on Heroku
- [x] Client deployed on Heroku/Azure/AWS

## Useful links
https://docs.github.com/en/actions/deployment/deploying-to-your-cloud-provider/deploying-to-azure/deploying-nodejs-to-azure-app-service

https://itnext.io/node-js-app-continuous-deployment-with-aws-pm2-and-github-actions-69a30ca29e9e

https://dev.to/stretch0/deploy-your-node-app-to-ec2-with-github-actions-h9a

### PR DESCRIPTION
**Pull request by dougUCN**
## Changes
- Adds github actions pipeline to deploy to AWS EC2 instance for staging
- Adds a copy of Nginx config file to repo for version controlling
- Add `npm run staging` command
- Enables client to get graphql endpoint url from a `.env` /`.env.local` type file
- Github actions now autogenerates a "deployments" section with a link to staging

Closes #8 

## To Do
- [x] Resolve CORS issue when AWS queries the Heroku endpoint (see dougUCN/LANE-server#14)
- [x] Add documentation on Nginx + PM2 and Amazon instance setup

### CODE REVIEW

File: src/main.tsx [hunk 1]
```diff
@@ -11,15 +11,22 @@ import { CompletedRun, EMS, LiveRun } from "components/EMS";
 import Navbar from "components/Navbar";
 import NotFound from "components/shared/NotFound";
 
+const GRAPHQL_WS_ENDPOINT =
+  import.meta.env.VITE_GRAPHQL_WS_ENDPOINT || "ws://localhost:8000/graphql/";
+
+const GRAPHQL_HTTP_ENDPOINT =
+  import.meta.env.VITE_GRAPHQL_HTTP_ENDPOINT ||
+  "http://localhost:8000/graphql/";
+
```
Comment (khyeh) [inline src/main.tsx:20]: I think we should modify this a little bit to give developers the option of pointing to either their local BE server or to heroku. That way, we can do FE development without ever having to spin up a local BE server.

We can do this by adding an .env.example file to git. That way, each developer can individually configure their own .env file without committing it to git.

https://vitejs.dev/guide/env-and-mode.html#env-files

Comment (dougUCN) [reply src/main.tsx:20]: This is exactly what I do on the AWS server (and this is already implemented in your highlighted code). The `deploy.sh` script creates a `.env.local` file with environment variables `VITE_GRAPHQL_HTTP_ENDPOINT` and `VITE_GRAPHQL_WS_ENDPOINT` that point to the heroku endpoint.

The logic in the code here simply states that if no vite env variables of those names are found, then it defaults to the localhost ports

I will update the readme

Comment (dougUCN) [reply src/main.tsx:20]: Readme updated in fbe906c71f7d16846c6b5f2b17e49fdcd2bf1b25

Comment (dougUCN) [reply src/main.tsx:20]: After further discussion, added `.env.example` file in 729aabf04c978991072d82b82191ad9422c1f5c5

**Review by khyeh — APPROVED** [review]

**Pull request merged**